### PR TITLE
System test

### PR DESCRIPTION
**Pull request by joshpuetz**
What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update
- [x] Test

## Description

Adds a system test for email login. That's it: that's the PR!

## Added tests?

- [x] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

test

### CODE REVIEW

**Review by jacobherrington — COMMENTED** [review]
This definitely needs a spec! Great catch :+1:

A couple of suggestions that might be good to take care of now.

**Review by citizen428 — APPROVED** [review]
👍 for a spec only PR! I'll approve this now, but I'd really like to see Jacob's suggestion re `allow(SiteConfig).to receive` implemented, our test suite is already a bit too flaky, we shouldn't add more potential causes for spec failuers.

**Review by benhalpern — COMMENTED** [review]
LGTM after committing @jacobherrington's suggestions

File: spec/system/authentication/user_logs_in_with_email_spec.rb [hunk 1]
```diff
@@ -0,0 +1,122 @@
+require "rails_helper"
+
+RSpec.describe "Authenticating with Email" do
+  let(:sign_in_link) { "Continue" }
+  let(:sign_up_link) { "Sign up with Email" }
+
+  before do
+    SiteConfig.allow_email_password_registration = true
```
[the remark this answers is not in the document]
Comment (joshpuetz) [reply spec/system/authentication/user_logs_in_with_email_spec.rb:8]: I can certainly make this change, but my question is _why_ is this better? Seems like an unnecessary mock to me.

[the remark this answers is not in the document]
Comment (joshpuetz) [reply spec/system/authentication/user_logs_in_with_email_spec.rb:8]: Ah..then I scrolled down and saw @citizen428's comment! (https://github.com/forem/forem/pull/10854#pullrequestreview-508900257). I'm direct use of `SiteConfig` is a risk to test flakiness, but I can roll with it!

**Review by jacobherrington — APPROVED** [review]

**Pull request merged**